napari/_qt/dialogs/qt_reader_dialog.py

```diff
@@ -252,5 +259,5 @@ def open_with_dialog_choices(
     if persist:
         get_settings().plugins.extension2reader = {
             **get_settings().plugins.extension2reader,
-            f'*{extension}': plugin_name,
```

Comment:
@ppwadhwa this is the reason why your test is failing - when saving a preference for a file extension we definitely want to add the `*`. However when saving the absolute path for a folder, we shouldn't have the `*`. This probably needs to go behind a check on whether the `extension` is actually an absolute path

Reply:
I actually added in the * earlier, so the * is already there.  I deal with the path check in lines 40-49.  Let me know if this is a problem.

Reply:
Ah I see, doing it in the `init`. It's definitely not a problem to do it there, but I think this is definitely what's causing the failing test, so we may need to look into this a bit more

Reply:
ok, I changed it around again so the test should pass now (it did locally, just assuming it will here too).